microprofile/tests/junit5/src/main/java/io/helidon/microprofile/tests/junit5/HelidonTest.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.microprofile.tests.junit5;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.junit.jupiter.api.extension.ExtendWith;
+
+/**
+ * An annotation making this test class a CDI bean with support for injection.
+ * <p>
+ * There is no need to provide {@code beans.xml} (actually it is not recommended, as it would combine beans
+ * from all tests), instead use {@link io.helidon.microprofile.tests.junit5.AddBean},
+ * {@link io.helidon.microprofile.tests.junit5.AddExtension}, and {@link io.helidon.microprofile.tests.junit5.AddConfig}
+ * annotations to control the shape of the container.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+@ExtendWith(HelidonJunitExtension.class)
+public @interface HelidonTest {
+    /**
+     * Whether discovery is automated or disabled.
+     * <p>
+     * When discovery is enabled, the whole classpath is scanned for bean archives (jar files containing
+     * {@code META-INF/beans.xml}) and all beans and extensions are added automatically.
+     * <p>
+     * When discovery is disabled, CDI would only contain the CDI implementation itself and beans and extensions added
+     * through annotations {@link io.helidon.microprofile.tests.junit5.AddBean} and
+     * {@link io.helidon.microprofile.tests.junit5.AddExtension}
+     *
+     * @return whether to do discovery, defaults to {@code true}
+     */
+    boolean discovery() default true;
```

Comment:
Would be nice to add something like `boolean dirtiesContext() default false;` if we want fresh container for every test

Reply:
Related question, how do JUnit extensions work if TestA runs before TestB and sets config, extensions and beans. Can TestB run with a fresh environment?

Reply:
Can be done by injecting beans to the test methods params instead

Reply:
That is quite expensive and complicated. Considering the test class itself is supposed to be a CDI bean, shutting the container and starting it again between tests would be unexpected (as you would need to run the next test method on a different instance of the bean)

Reply:
I will try to do the following and see how complex it is:
Allow `@HelidonTest(resetPerTest="true")`

In such a case:
1. No injection allowed into the class itself or using constructor
2. Annotations would be combined from class level and method levels
3. Each method would run in a separate container
4. Test methods can have `SeContainer` as a parameter

Reply:
Awesome thats exactly it! Works like a charm!
```java
    private WebClient client;
    private TestMessagingBean bean;

    @BeforeEach
    void beforeEach() {
        ServerCdiExtension server = CDI.current().select(ServerCdiExtension.class).get();
        bean = CDI.current().select(TestMessagingBean.class).get();
        client = WebClient.builder()
                .baseUri("http://localhost:" + server.port())
                .addReader(JsonpSupport.reader())
                .build();
    }

    @Test
    void alivenessWithErrorSignal(SeContainer seContainer) {
        TestMessagingBean bean = seContainer.select(TestMessagingBean.class).get();
```

Reply:
And its faster!
Container per method coded manually:
![image](https://user-images.githubusercontent.com/1773630/93305520-e725f600-f7fe-11ea-86f2-f825703b9607.png)
Container per method with `@HelidonTest(resetPerTest = true)`:
![image](https://user-images.githubusercontent.com/1773630/93305626-0cb2ff80-f7ff-11ea-89b7-29ef8a9281ef.png)